cmake/modules/RootConfiguration.cmake

```
@@ -635,6 +635,22 @@ else()
    set(has_found_attribute_noinline undef)
 endif()
 
+# We could just check `#ifdef __cpp_lib_hardware_interference_size`, but on at least Mac 11
+# libc++ defines that macro but is missing the actual feature
+# (see https://github.com/llvm/llvm-project/commit/174322c2737d699e199db4762aaf4217305ec465).
+# So we need to "manually" check instead.
+# `#ifdef R__HAS_HARDWARE_INTERFERENCE_SIZE` could be substituted by `#ifdef __cpp_lib_hardware_interference_size`
+# when Mac 11's life has ended (assuming the libc++ fix makes it in the next MacOS version).
+CHECK_CXX_SOURCE_COMPILES("
+#include <new>
+inline std::size_t constexpr foo = std::hardware_destructive_interference_size;
```

I think rather than relying on the compiler to be able to compile inline variables I'd rather use
```suggestion
using Check_t = char[std::hardware_destructive_interference_size];
```
which also ensures that `std::hardware_destructive_interference_size` is a constant expression.

---

done (although `inline constexpr` also ensures the constexpr-ness I think)